src/components/ClaimFilterWasCategorised.js

```diff
@@ -59,16 +60,27 @@ class ClaimFilterWasCategorised extends Component {
     }
 
     _checkAddCategorization(json_ext, prev_json_ext) {
-        var json_exists = (json_ext === null );
-        var valueUpdated = (prev_json_ext['value']['wasCategorized'] != this.state.categorized && this.state.categorized === true);
-        return (json_exists || valueUpdated)
-        
+        const { filters } = this.props;
+        var json_not_exists = (json_ext === null);
+        var prev_claim_param = prev_json_ext['value']
+        var valueExisted = !!prev_claim_param['claim_ai_quality']
+        var valueUpdated = false
+        var wasCategorized = false
+        if (valueExisted) {
+            wasCategorized = prev_claim_param['claim_ai_quality']['was_categorized']
+        }
+        console.log("Was cat: " + wasCategorized)
+        var valueUpdated = (wasCategorized != this.state.categorized && this.state.categorized === true);
+
+        var statusChecked = (!!filters['claimStatus'] && filters['claimStatus']['value'] === 4)
+
+        return statusChecked && (json_not_exists || valueUpdated)  
     }
 
     _checkRemoveJson(json_ext) {
         var checkboxState = this.state.categorized === false;
-        var jsonHasAttribute = !!json_ext['value']['wasCategorized']
-
+        var jsonHasAttribute = (!!json_ext && !!json_ext['value']['claim_ai_quality'])
+        console.log(JSON.stringify(json_ext), json_ext['value']['claim_ai_quality'])
```

Comment:
I suppose this is for debugging. Please remove if not required.

Reply:
Removed